src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -362,8 +362,32 @@ fun processPathResolveVariants(lookup: ImplLookup, path: RsPath, isCompletion: B
             if (processor("self", base)) return true
         }
         if (processItemOrEnumVariantDeclarations(base, ns, processor, withPrivateImports = isSuperChain)) return true
-        if (base is RsTypeDeclarationElement && parent !is RsUseSpeck) {
-            // Foo::<Bar>::baz
+        if (base is RsTypeDeclarationElement && parent !is RsUseSpeck) { // Foo::<Bar>::baz
+            val shadowingProcessor = run {
```

Comment:
Maybe we want to extract this into separate function? Because path resolution is already too big

Reply:
I don't sure this peace of code is good for extraction. We can extract large branches (in a separate PR) instead: where `if (qualifier != null)`, `if (typeQual != null)` and remained